src/AppInstallerRepositoryCore/Microsoft/Schema/1_4/DependenciesTable.cpp

```diff
@@ -137,7 +137,12 @@ namespace AppInstaller::Repository::Microsoft::Schema::V1_4
             {
                 insert.Reset();
                 insert.Bind(1, dep.m_manifestRowId);
-                insert.Bind(2, dep.m_versionRowId.has_value() ? std::to_string(dep.m_versionRowId.value()) : "");
+
+                if (dep.m_versionRowId.has_value())
```

Comment:
I'm not exactly sure on what the bind state is after a `Reset`, but I think it doesn't change.  You want to `Bind(2, nullptr)` to set the value to null.

Also note that you have to use `GetColumnIsNull` when reading the rows to check for a DB null value before you can `GetColumn<std::string>`.